doc/source/Code-Style-Guide.rst

```diff
@@ -0,0 +1,746 @@
+                   A multiplier.
+
+           Returns:
+               The multiplied value.
+
+           Raises:
+               ValueError: If the value exceeds the allowed
+                   upper limit of 100.
+               ValueError: If the value is below or equal to
+                   zero.
+
+           Warns:
+               Warning: If the value is outside of the
+                   expected range (> 0 and <= 50).
+
+           References:
+               Bauer, P., Thorpe, A., Brunet, G. (2015) The quiet
+                revolution of numerical weather prediction.
+               Nature, Vol 525, pp 47-55
+           """
+           updated_simple_variable = (self.simple_variable * 2) / 3
+           if updated_simple_variable > 100:
+               msg = (
+                   "An updated simple variable of {} exceeds "
+                   "the allowable upper limit of 100.".format(updated_simple_variable)
+               )
+               raise ValueError(msg)
+           elif updated_simple_variable > 50:
+               msg = (
+                   "The updated simple variable of {} "
+                   "is higher than expected. "
+                   "Expected range is > 0 and <= 50.".format(updated_simple_variable)
+               )
+               raise warnings.warn(msg)
+           elif updated_simple_variable <= 0:
+               msg = (
+                   "An updated simple variable of {} is "
+                   "below the allowable lower limit of 0.".format(updated_simple_variable)
+               )
+               raise ValueError(msg)
+           return updated_simple_variable
+
+       def process(self, cube: Cube) -> Cube:
+           """Description for what's done in the process method.
+
+           Args:
+               cube:
+                   An example cube for processing.
+
+           Returns:
+               Output cube after multiplying the input cube by the
+               simple variable.
+           """
+           # Inline comments can be added, if required.
+           cube = self.my_static_method(cube)
+           (max_val, min_val) = self.my_static_method_multiple_returns(cube)
+           new_cube = cube
+           if min_val > 0.0 and max_val < 100.0:
+               new_simple_variable = self.my_method()
+               new_cube = cube * new_simple_variable
+           return new_cube
+
+Helper Functions/Methods
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Helper functions or methods may live in one of several places. Where
+they should live depends on how they will be used.
+
+**Case 1:** A function used in more than one module. - In this case the
+function should be located in a shared location e.g. utilities.py
+
+**Case 2:** A function used by several classes within one module. - The
+function should be kept outside of any one class, but within the module.
+
+**Case 3:** A function used only within a single class. - Should be kept
+within the class; as a static method if it makes no use of self.
```

Comment:
More bullets that has been inlined.
https://bens-improver-fork.readthedocs.io/en/wikitest/Code-Style-Guide.html#helper-functions-methods

Reply:
Fixed.